### PR TITLE
Don't prune traversal by leaf vertex subplans

### PR DESCRIPTION
**Pull request by milroy**
Issue #1260 and logs from production clusters have reported match errors such as ```by_subplan: planner_multi_avail_during returned -1.``` for JGFs with `node` vertices with an edge to the `cluster` vertex. The error occurs because PR #1248 prevents creation of subplan filters on leaf vertices, causing subsequent checks against the planner to fail. 

This PR adds a check to `by_subplan` to bypass pruning based on null subplans.

Currently, the test case succeeds without the fix because the error isn't logged in `flux dmesg` for some reason.

### CODE REVIEW

File: resource/traversers/dfu_impl.cpp [hunk 1]
```diff
@@ -176,6 +176,11 @@ int dfu_impl_t::by_subplan (const jobmeta_t &meta,
     int saved_errno = errno;
     planner_multi_t *p = (*m_graph)[u].idata.subplans[s];
 
+    if (!p) {
+        // Subplan is null if u is a leaf.
```
Comment (jameshcorbett) [inline resource/traversers/dfu_impl.cpp:180]: Could the subplan be NULL for any other reasons? Are there any more checks needed here?

**Review by jameshcorbett — COMMENTED** [review]
Looks good although it would be good to make sure the test fails without the PR...

File: resource/traversers/dfu_impl.cpp (hunk 1 above)
Comment (trws) [reply resource/traversers/dfu_impl.cpp:180]: We discussed this in person this afternoon. If it's null for any other reason, it's unintentional, but checking if the vertex is a leaf or not is an unfortunately tricky/expensive operation to use in something that gets called this much.

**Comment by milroy** [comment]
After many attempts I was unsuccessful at getting the error to propagate to `flux dmesg`. I updated the testsuite test to use `resource-query` instead. Let me know if I should keep poking at the resource module-based test.

I think the PR is ready for review.

**Review by jameshcorbett — APPROVED** [review]
LGTM thanks for this!

File: resource/traversers/dfu_impl.cpp (hunk 1 above)
Comment (milroy) [reply resource/traversers/dfu_impl.cpp:180]: Noting for posterity that I tested pruning filter configurations that skip intermediate vertices (e.g., `cluster:core, node:core`, to skip `rack` vertices) and observed that the `rack` vertex subplan is not null. Apparently the code creates the planners in the intermediate vertices anyway.

**Comment by milroy** [comment]
Thanks for the reviews. Setting MWP.

**Pull request merged**